Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

gcc 7 build issue

--- a/Zend/zend_language_parser.c
+++ b/Zend/zend_language_parser.c
@@ -40,6 +40,11 @@
    define necessary library symbols; they are noted "INFRINGES ON
    USER NAME SPACE" below.  */
 
+/* php cores dump when built with gcc 7 on sparc with -O2 and above */
+#ifdef __sparc__
+#pragma GCC optimize ("-fno-gcse")
+#endif
+
 /* Identify Bison output.  */
 #define YYBISON 1
 
--- a/aclocal.m4	2018-05-29 19:03:22.061223379 +0000
+++ b/aclocal.m4	2018-05-29 19:03:35.766611671 +0000
@@ -2249,8 +2249,7 @@
     PHP_EVAL_INCLINE($ICU_INCS)
     PHP_EVAL_LIBLINE($ICU_LIBS, $1)
 
-    ICU_CXXFLAGS=`$ICU_CONFIG --cxxflags`
-    ICU_CXXFLAGS="$ICU_CXXFLAGS -DU_USING_ICU_NAMESPACE=1"
+    ICU_CXXFLAGS="-DU_USING_ICU_NAMESPACE=1"
   fi
 ])
 
